### DISCUSSION TITLE
Problem with @db.Time(0)

### DISCUSSION THREAD

**Discussion by hckhanh** [prisma/prisma#18306] [Q&A]
I have a schema like this (PostgreSQL):

```prisma
model Campaign {
  id         String    @id @default(cuid())

  start_date DateTime? @db.Date
  start_time DateTime? @db.Time(0)
  end_date   DateTime? @db.Date
  end_time   DateTime? @db.Time(0)
}
```

When I try to query campaigns with `start_time`, it won't work at all, even though the value is correct

```ts
const current = new Date('1969-12-31T23:18:00.564Z');
await prisma.campaign.findMany({
        where: {
          start_time: { lte: current },
        },
      })
```

In my case, the `start_time` in db is `1970-01-01T00:00:00.000Z` and the current time is `1969-12-31T23:18:00.564Z`
I think there is a bug here.

**Comment by nurul3101** [comment]
Hello @hckhanh 👋 

Based on your schema file, I tried to reproduce the issue but it works as expected for me.
Can you try this script?

```typescript
import { PrismaClient, Prisma } from '@prisma/client';

const prisma = new PrismaClient();

async function main() {
  await prisma.campaign.deleteMany();

  await prisma.campaign.create({
    data: {
      start_date: new Date('1970-01-01T00:00:00.000Z'),
      end_date: new Date('1970-01-01T00:00:00.000Z'),
      start_time: new Date('1970-01-01T00:00:00.000Z'),
      end_time: new Date('1970-01-01T00:00:00.000Z'),
    },
  });

  const findCampaign = await prisma.campaign.findMany({
    where: {
      start_time: {
        lte: new Date('1969-12-31T23:18:00.564Z'),
      },
    },
  });

  console.log(findCampaign);
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```

Response:
<img width="388" alt="Screenshot 2023-03-14 at 12 47 30 PM" src="https://user-images.githubusercontent.com/13049676/224924301-e7176705-360f-4242-8904-e69d346a1333.png">

**Reply by hckhanh** [reply]
Hi @nurul3101, I have a calculation for myself

```ts
const start_time = new Date('1970-01-01T00:00:00.000Z');
const current = new Date('1969-12-31T23:18:00.564Z');

const diff = start_time - current // 2519436
```

So the `diff` value is greater than `0`, which means `start_time` must be greater than `current`.

And I think that if you get any result from this query, it must be incorrect. The query is supposed to look for the records that have the `start_time` <= `current`

```ts
const findCampaign = await prisma.campaign.findMany({
    where: {
      start_time: {
        lte: new Date('1969-12-31T23:18:00.564Z'),
      },
    },
  });
```

**Reply by nurul3101** [reply]
@hckhanh In the example that I shared the current time was `new Date('1969-12-31T23:18:00.564Z')` and the start time was `new Date('1970-01-01T00:00:00.000Z')` so current time was less than the start time. So Ideally, this script shouldn't have returned any records
```ts
  const findCampaign = await prisma.campaign.findMany({
    where: {
      start_time: {
        lte: new Date('1969-12-31T23:18:00.564Z'),
      },
    },
  });
```
But it does, so it looks like a bug. Did I understand this correct?

**Reply by hckhanh** [reply]
Yes @nurul3101, that is what I mean.

**Reply by nurul3101** [reply]
Do you mind creating a [Bug Report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml)? Our engineering team monitors that and can confirm if this indeed is a bug or if there is something else which we are missing.

**Reply by hckhanh** [reply]
I had created a issue referred to this discussion: https://github.com/prisma/prisma/issues/18375
